### ISSUE TITLE
ORCID plugin should not function without credentials entered

### ISSUE THREAD

**Issue by librariam** [pkp/orcidProfile#177]
**Is your feature request related to a problem? Please describe.**
Here is the most common scenario and a source of 99% of ORCID-related tickets filed with PKP|PS:

- A JM/JE enables the plugin without obtaining/entering credentials
- Authors start receiving ORCID emails with sandbox URL which does not work
- JM/JE report this as a tech issue with the plugin
- Following the explanation & instructions on plugin setup most give up on it altogether

Unfortunately I find that this initial experience sours the first impression and turns people away from using the plugin. 

**Describe the solution you'd like**
The bottom line is that the plugin should not be performing any of its functionality (particularly not direct author interactions) without being set up first. This follows the logic of #161 

Some ideas for implementation:
- an additional checkbox within the plugin settings that will not become active until credentials are entered and validated
- a pop up when enabling the plugin that alerts the JM/JE that it needs to be set up + link to the ORCID guide
- disable the plugin unless credentials are entered

**Comment by asmecher** [comment]
@withanage, this would be a good improvement to include in the next release.

**Comment by withanage** [comment]
@librariam @asmecher 

sure, I will add the  following settings into the next release.

1.     an additional checkbox within the plugin settings that will not become active until credentials are entered and validated

1.     disable the plugin unless credentials are entered



1. a pop up when enabling the plugin that alerts the JM/JE that it needs to be set up + link to the ORCID guide

@asmecher , is this generally possible using a hook mechanism currently?

**Comment by asmecher** [comment]
@withanage, just off the top of my head, the PLN plugin had some conditions that were required in order for the plugin to be enabled:
https://github.com/pkp/pln/blob/ojs-stable-3_1_2/PLNPlugin.inc.php#L375

**Comment by withanage** [comment]
Hi @librariam ,

I have implemented the plugin as per your definition, before releasing it and doing some more testing, I wanted to ask, if the implementation corresponds to your requirements.

Here is the workflow.
1.  At the beginning plugin is disabled.
1.  Settings is available  _also_ when the plugin is disabled.
1. Status shows about the status of the plugin. (I will link the plugin doc here)
1. With   wrong.-credentials you cannot enable the plugin.
1. With  correct credentials you can enable the plugin.
1.  If globally configured the  same rules for client validation apply and the save button is disabled.

One thing, which I could not add was  a separate pop-up. But the plugin is not  allowed to be enabled as per definition 2.

![orcid_validation](https://user-images.githubusercontent.com/1921992/127024216-c20b576b-615c-4453-9c71-be4c19016dd3.gif)

**Comment by librariam** [comment]
Thank you @withanage . Just to clarify - what would be the behaviour if someone tries to enable the plugin without entering credentials?

**Comment by withanage** [comment]
@librariam  , users cannot enable the plugin, until they enter credentials (valid) through the backend.  They can click the enable plugin, but it will not enable it.

**Comment by librariam** [comment]
@withanage thank you, this is great! Not being able to enable the plugin without entering valid credentials is exactly what we're after!

**Comment by withanage** [comment]
@librariam 

I have released this enhancements for 

- OJS/OPS 3.3
https://github.com/pkp/orcidProfile/tree/v1_1_2-21

- OJS/OPS 3.2
 https://github.com/pkp/orcidProfile/releases/tag/v1_1_2-20

and integrated in the plugin-gallery.

**Comment by asmecher** [comment]
@withanage, is this issue ready to be closed?

**Comment by withanage** [comment]
@asmecher    There was no bugs  reported  after adding the functionality in around 2 weeks.  Yes,  I will close it.
